src/ast/nodes/BlockStatement.ts

```diff
@@ -15,7 +15,7 @@ export default class BlockStatement extends StatementBase {
 	type: NodeType.tBlockStatement;
 	body: StatementNode[];
 
-	bindImplicitReturnExpressionToScope() {
+	addImplicitReturnExpressionToScope() {
 		const lastStatement = this.body[this.body.length - 1];
 		if (!lastStatement || lastStatement.type !== NodeType.ReturnStatement) {
 			this.scope.addReturnExpression(UNKNOWN_EXPRESSION);
```

Comment:
There originally was an `UNDEFINED_EXPRESSION` but it turned out to be equivalent to `UNKNOWN_EXPRESSION` so I dropped it. With the new literal value tracking, re-introducing it would actually make sense. Since this is intended to be a bug-fix, I would postpone this for a PR explicitly addressing #2223.